### PR TITLE
[SPARK-28957][SQL] Copy any "spark.hive.foo=bar" spark properties into hadoop conf as "hive.foo=bar"

### PR DESCRIPTION
**Pull request by WeichenXu123**
### What changes were proposed in this pull request?

Copy any "spark.hive.foo=bar" spark properties into hadoop conf as "hive.foo=bar"

### Why are the changes needed?
Providing spark side config entry for hive configurations.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
UT.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
Jenkins retest this please.

**Review by felixcheung — COMMENTED** [review]
probably ok... I was a bit worry that existing app having spark.hive.* might now break, but probably low chance that is happening?

**Comment by WeichenXu123** [comment]
@cloud-fan What do you think ?
Should we override existing `spark.hive` config ?

**Comment by cloud-fan** [comment]
I like this proposal. It's much better than looking at all hive configs and re-implementing it in Spark. (see how `InsertIntoHiveTable` handles `"hive.exec.dynamic.partition` and `hive.exec.dynamic.partition.mode`).

@WeichenXu123 can you resolve the conflicts?

**Comment by cloud-fan** [comment]
Can we also update `configuration.md#custom-hadoophive-configuration`?

**Comment by cloud-fan** [comment]
thanks, merging to master!

File: core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala [hunk 1]
```diff
@@ -79,8 +79,10 @@ private[spark] class SparkHadoopUtil extends Logging {
    * Appends S3-specific, spark.hadoop.*, and spark.buffer.size configurations to a Hadoop
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala]: nit. This function description seems to become behind, but, PR looks good.
Thanks!

**Pull request closed without merging**